src/content/docs/controller.md

```diff
@@ -2,4 +2,218 @@
+
+Joystick states can be checked using:
+
+-   `left_stick` (for the left joystick)
+-   `right_stick` (for the right joystick)
+
+## Button states
+
+To check whether a button is _currently pressed down_, you can use the `is_pressed` method on the button. This method returns `true` if the button is pressed and `false` otherwise.
+
+> Wait, but I want to know when a button was _just_ pressed!
+
+Don't worry! In VEXCode, you may have had to declare a separate variable just to keep track of the previous state of a button. In vexide, we do this for you!
+
+You can use the `is_now_pressed` method to check if a button was just pressed. This method returns `true` if the button was pressed in the last update (i.e., it was released the last time `Controller::state` was called and is now pressed) and `false` otherwise. This is quite useful for implementing actions like toggling a pneumatic piston or changing the state of a subsystem.
+
+> [!NOTE]
+> If a button was quickly pressed and released in between two updates, the press will not be detected by `is_now_pressed` and will be dropped. Additionally, the controller's update interval is 25ms, so the minimum time between two updates is 25ms.
+
+```rs
+// @fold start
+#![no_std]
+#![no_main]
+
+use vexide::prelude::*;
+
+// @fold end
+#[vexide::main]
+async fn main(peripherals: Peripherals) {
+  let mut controller = peripherals.primary_controller;
+  let mut piston = DigitalOut::new(peripherals.port1);
+
+  loop {
+    let state = controller.state().unwrap_or_default();
+    //                             ^
+    // [If the controller is disconnected or there's a problem, we'll just use the default state.]
+
+    //                (              )
+    if state.button_a.is_now_pressed() {
+    //                ^
+    //         [Returns true if the A button was just pressed]
+      piston.toggle();
+    }
+
+    sleep(Controller::UPDATE_INTERVAL).await;
+  }
+}
+```
+
+## Joystick states
+
+The VEX controller has two joysticks: one on the left and one on the right. You can access the state of these joysticks using the `left_stick` and `right_stick` properties on the `ControllerState` struct, respectively.
+
+For fun, let's try implementing a simple tank drive program using the controller's joysticks. We'll map the y-axis of the left joystick to the left motor and the y-axis of the right joystick to the right motor. Notice that `JoystickState::x` and `JoystickState::y` return values in the interval `[-1, 1]`, and not a percentage like VEXCode does.
+
+```rs
+// @fold start
+#![no_std]
+#![no_main]
+
+use vexide::prelude::*;
+
+// @fold end
+#[vexide::main]
+async fn main(peripherals: Peripherals) {
+  let mut controller = peripherals.primary_controller;
+  let mut left_motor = Motor::new(peripherals.port1);
+  let mut right_motor = Motor::new(peripherals.port2);
+
+  loop {
+    let state = controller.state().expect("Failed to get controller state");
```

Comment:
Should be unwrap_or_default

Reply:
Oops, I only fixed it once. Thanks for reviewing!